docs/proposals/20220923-nodeselector.md

```diff
@@ -0,0 +1,141 @@
+---
+title: Adding Global Node Selector in Pod Syncer
+authors:
+- "@m-messiah"
+  reviewers:
+- "@christopherhein"
+- "@Fei-Guo"
+- "@weiling61"
+  creation-date: 2022-09-23
+  last-updated: 2022-09-23
+  status: provisional
+  see-also: []
+  replaces: []
+  superseded-by: []
+---
+
+# Adding Global Node Selector in Pod Syncer
+
+## Table of Contents
+
+<!--ts-->
+
+* [Adding Quota Validation Plugin in Pod Syncer]
+    * [Table of Contents](#table-of-contents)
+    * [Glossary](##glossary)
+    * [Summary](##summary)
+    * [Motivation](##motivation)
+        * [Goals](###goals)
+        * [Non-Goals](###non-goalsk)
+    * [Proposal](##proposal)
+        * [User Stories and Requiremments](###user-stories-and-requirements)
+        * [Implementation Details](###implementation-details)
+        * [Security Model](###security-model)
+        * [Risks and Mitigations](###risks-and-mitigations)
+    * [Alternatives](##alternatives)
+    * [Implementation History](##implementation-history)
+
+
+<!--te-->
+
+## Glossary
+
+Refer to the https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/main/docs/proposals/00_capn-glossary.md
+
+## Summary
+
+NodeSelector is a widely used mechanism to assign pods to specific node groups providing isolation and predictability of schedule. Current Multi-tenant Virtual Cluster implementation does not have any control over it. In this proposal, we suggest having a VirtualClusterNodeSelector object to tell PodSyncer to apply nodeSelector to every pod created in the virtual cluster.
+
+## Motivation
+
+- The current implementation does not allow to have a fully-isolated virtual cluster, by assigning its workload to the specific nodes without any knowledge about superCluster
+- Operators of the VirtualCluster can use nodeSelectors right now, but they have no visibility of what labels exist in the superCluster
+- Operators of the VirtualCluster have no control over control-plane scheduling, which should be isolated on the same nodes too
+
+### Goals
+
+- Propose VirtualClusterNodeSelector isolation mechanism for a user to integrate it into pod-syncer and cluster provisioner
+- Define the sufficient structure of the VirtualClusterNodeSelector object to control pod-syncer and cluster provisioner behaviour and cover all pod scheduling cases
+- Implement pod-syncer and cluster provisioner mechanisms that respect the VirtualClusterNodeSelector object and modify pod scheduling.
+
+### Non-Goals/Future Work
+
+- Super Cluster node labelling and aggregation are not covered here, and it is assumed that we already have nodes with different labels in the supercluster
+- Usage of different schedulers is covered in different docs
+
+## Proposal
+
+### User Stories
+
+- As a cluster admin, I want to allow tenants (if they need to) to have isolated virtual clusters and isolate them from any other workload (for performance, security or special needs)
```

Comment:
Also, node selecting spec and future pod spec may not fully match during provisioning time, will this ends up that tenant cannot run certain pods and they are not aware of it?

Reply:
> If one tenant defines the node selector, will this mean all other tenant needs to aware of the constrain? otherwise, other tenant may schedule pod to the node selected by the first tenant? Will this cause interference of one tenant to all other tenants?

I don't think it is a problem if we allow them to use the same nodes, since superCluster scheduler still has its quota handling and checks. Even more, we could posibly have a tenant, who would have more than one virtualcluster, but wants to be settled in the same group of nodes